frontend/src/app/accountCreation/PasswordForm/PasswordForm.tsx

```diff
@@ -62,36 +75,58 @@ export const PasswordForm = () => {
     return "";
   };
 
+  // This function set the hint on the page, then returns the number of passing requirements
   const calculateStrength = (value: string): number => {
     setStrengthHint(buildHint(value));
     return matchedRequirements(value).length;
   };
 
-  const handleChange = ({
+  // onChange handler for password field
+  const handlePasswordChange = ({
     target: { value },
   }: ChangeEvent<HTMLInputElement>) => {
     setPassword(value);
     setStrength(calculateStrength(value));
   };
 
-  const validatePassword = () => {
+  // Clears the hint and sets the error since this is validation on submit
+  const validatePassword = (): boolean => {
     setStrengthHint("");
-    setPasswordError(buildHint(password));
+    const hint = buildHint(password);
+    setPasswordError(hint);
+    return hint === "";
   };
 
-  const validatePasswordConfirmation = () => {
+  const validatePasswordConfirmation = (): boolean => {
+    let error = "";
     if (password !== passwordConfirmation) {
-      setPasswordConfirmationError("Passwords must match");
-    } else {
-      setPasswordConfirmationError("");
+      error = "Passwords must match";
     }
+    setPasswordConfirmationError(error);
+    return error === "";
   };
 
-  const handleSubmit = () => {
-    validatePassword();
-    validatePasswordConfirmation();
+  // Form submit handler
+  const handleSubmit = async () => {
+    if (validatePassword() && validatePasswordConfirmation()) {
+      setLoading(true);
+      let success = false;
+      try {
+        await AccountCreationApi.setPassword(activationToken, password);
+        success = true;
+      } catch (error) {
+        setPasswordError(`API Error: ${error}`);
+      } finally {
+        setLoading(false);
+        if (success) {
+          setSubmitted(true);
```

Comment:
You might be able to eliminate the `success` variable by putting this in the `try` block after the `await` line